lib/block-supports/typography.php

```diff
@@ -105,16 +105,21 @@ function gutenberg_apply_typography_support( $block_type, $block_attributes ) {
 	}
 
 	if ( $has_font_family_support ) {
-		$has_font_family = isset( $block_attributes['style']['typography']['fontFamily'] );
-		if ( $has_font_family ) {
-			$font_family = $block_attributes['style']['typography']['fontFamily'];
-			if ( strpos( $font_family, 'var:preset|font-family' ) !== false ) {
-				// Get the name from the string and add proper styles.
-				$index_to_splice  = strrpos( $font_family, '|' ) + 1;
-				$font_family_name = substr( $font_family, $index_to_splice );
-				$styles[]         = sprintf( 'font-family: var(--wp--preset--font-family--%s);', $font_family_name );
+		$has_named_font_family  = array_key_exists( 'fontFamily', $block_attributes );
+		$has_custom_font_family = isset( $block_attributes['style']['typography']['fontFamily'] );
+
+		if ( $has_named_font_family ) {
+			$classes[] = sprintf( 'has-%s-font-family', gutenberg_experimental_to_kebab_case( $block_attributes['fontFamily'] ) );
+		} elseif ( $has_custom_font_family ) {
+			// Before using classes, the value was serialized as a CSS Custom Property.
+			// We don't need this code path when it lands in core.
+			$font_family_custom = $block_attributes['style']['typography']['fontFamily'];
+			if ( strpos( $font_family_custom, 'var:preset|font-family' ) !== false ) {
+				$index_to_splice  = strrpos( $font_family_custom, '|' ) + 1;
+				$font_family_slug = substr( $font_family_custom, $index_to_splice );
+				$classes[]        = sprintf( 'has-%s-font-family', $font_family_slug );
```

Comment:
Do we need `gutenberg_experimental_to_kebab_case` here?

```suggestion
				$classes[]        = sprintf( 'has-%s-font-family', gutenberg_experimental_to_kebab_case( $font_family_slug ) );
```

Reply:
This was actually a bug in the old code path. I've added tests for the three code paths and fixed this at https://github.com/WordPress/gutenberg/pull/31910/commits/19dbeee39b119ec8e814e7b5bd9200f4798e9a3d